Actor.py

```diff
@@ -195,74 +209,69 @@ def follow_path(self, delta_time, sim_time, path, time_to_collision=None, collis
         if path:
             # Which path node have we most recently passed
             node_checkpoint = 0
-            
-
             # Ideally we should first calculate acceleration, then velocity, then position (euler integration)
             # For now, we'll ignore acceleration
             # TODO: This could be improved by saving the current path node instead of having to find it again every tick
             # Calculate velocity
+
+            
             for i in range(len(path)-1):
                 n1 = path[i]
                 n2 = path[i+1]
+
                 if (n1.s <= self.state.s <= n2.s):
 
                     node_checkpoint = i
 
                     # if collision point provided, use ensured collision logic
-                    
                     if path and time_to_collision is not None and collision_segment_prev_node is not None and collision_segment_next_node is not None:
-                        if self.type == 4 and not self.released and self.id != 1:
-                            
-                            # Project collision point to arc lengths
-                            diff = np.array(collision_pt) - np.array([collision_segment_prev_node.x,
-                                                                    collision_segment_prev_node.y])
-                            euclidian_dist = float(np.sqrt(np.sum(diff**2)))
-                            collision_pt_s = collision_segment_prev_node.s + euclidian_dist
+                        # Project collision point to arc lengths
+                        diff = np.array(collision_pt) - np.array([collision_segment_prev_node.x,
+                                                                collision_segment_prev_node.y])
+                        euclidian_dist = float(np.sqrt(np.sum(diff**2)))
+                        collision_pt_s = collision_segment_prev_node.s + euclidian_dist
 
-                            # Distance this oncoming vehicle must travel to the collision point (along s)
-                            distance_remaining = collision_pt_s - self.state.s
+                        # Distance this oncoming vehicle must travel to the collision point (along s)
+                        distance_remaining = collision_pt_s - self.state.s
 
+                        if self.type == 4 and not self.released and self.id != 1:
```

Comment:
what does the `4` refer to? The types are defined in subclasses
```
class Pedestrian(Actor):
    #pedestrian types
    PP_TYPE = 2    
    ...
    SP_TYPE = 4
```
and 
```
class Vehicle(Actor):
    #vehicle types
    ...
    PV_TYPE = 4
```
As you can see, just checking `== 4` is not correct for pedestrian, `PP = 2` not `4`. Also, please add a comment saying which constants these numbers are coming from.